scriptoocloud/src/test/java/edu/asu/diging/scriptoocloud/core/service/impl/GitRepositoryManagerServiceTest.java

```diff
@@ -0,0 +1,73 @@
+package edu.asu.diging.scriptoocloud.core.service.impl;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.net.MalformedURLException;
+import java.util.ArrayList;
+import java.util.Optional;
+
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.mockito.MockitoAnnotations;
+
+import edu.asu.diging.scriptoocloud.core.data.GitRepositoryRepository;
+import edu.asu.diging.scriptoocloud.core.model.GitRepository;
+import edu.asu.diging.scriptoocloud.core.model.impl.GitRepositoryImpl;
+import edu.asu.diging.scriptoocloud.core.service.DeleteFilesService;
+import edu.asu.diging.scriptoocloud.core.service.JgitService;
+import edu.asu.diging.scriptoocloud.core.service.UrlFormatterService;
+
+class GitRepositoryManagerServiceTest {
+
+    @InjectMocks
+    private GitRepositoryService serviceToTest;
+    
+    @Mock
+    private JgitService jGitService;
+    
+    @Mock
+    private GitRepositoryRepository gitRepositoryJpa;
+    
+    @Mock
+    private DeleteFilesService deleteFilesService;
+    
+    @Mock
+    private UrlFormatterService urlFormatter;
+    
+    @Mock
+    private GitRepositoryImpl gitRepository;
+    
+    @BeforeEach
+    public void init(){
+        MockitoAnnotations.initMocks(this);   
+    }
+    
+    @Test
+    public void test_cloneRepository_clone() throws MalformedURLException {
+        Mockito.when(urlFormatter.urlToFolderName("https://github.com/diging/scriptoocloud"))
+            .thenReturn("github.com_diging_scriptoocloud");
```

Comment:
I would have  the formatter service replace `.` with `_`. Some os might interpret it as an extension otherwise and make it harder to work with (not a huge deal, but might as well, since you have control over that).